### ISSUE TITLE
Release Twisted 16.0

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8210]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8210|
|Type|task|
|Created|2016-02-16 00:58:08Z|
|Branch|https://github.com/twisted/twisted/tree/release-16.0.0-8210|

toot toot no stopping the release train

<details><summary>Searchable metadata</summary>

```
trac-id__8210 8210
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__Twisted_16_0 Twisted 16.0
branch__branches_release_16_0_0_8210 branches/release-16.0.0-8210
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__None None
time__1455584288350303 1455584288350303
changetime__1458026687255475 1458026687255475
version__None None
owner__hawkowl hawkowl
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [46803]) Branching to release-16.0.0-8210.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

Builders running, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Thanks for the release branch.

Some tests are failing :(

I don't fully understand what is released here.

Why for example this branch contains changes for twisted/internet/test/test_endpoints.py and twisted/cred/checkers.py

It looks like this branch also deprecated twisted.names.version and similar public API. Is it ok?
I was expecting to deprecate them in a separate ticket and have this deprecation as part of the release notes.

Please let me know how I should review this branch.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl removed owner|
|-|-|

Hi Adi,

- The tests are failing due to OpenSSL 1.0.2f being installed on the builders. The 1.0.2f fixes were after this release was cut.

- This is a snapshot of Feb 16's trunk, with four cherry-picked commits made to trunk since that fix regressions in the 16.0 prereleases. If you look at the commit list in https://github.com/twisted/twisted/compare/trunk...release-16.0.0-8210, you can see which ones were cherrypicked, and how all code changes are from those cherry-picked commits.

- The deprecations and such are in the release notes, search for [#8219](https://github.com/twisted/twisted/issues/8219).

So, all that really needs reviewing is that the release notes are well formed, all of the cherry-picked commits have already been tested + reviewed, and are just there so that 16.0 releases in a good state.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

Thanks for the info. I was confused by the git diff as I was not expecting to see them in the diff.

------

Shouldn't this release also cherry pick the fix for OpenSSL 1.0.2f?

Does it make sense to release based on a 1 month old snapshot?

I don't plan to use Twisted 16.0, but if I was forced to upgrade to a newer Twisted version now, I would just went for trunk. But maybe is just my and for all the other people it make sense to have a release based on such an old version.

------------

I would say that changing conch.ssh from pycrypto to python cryptography is a big deal and it should be advertise as a top note.

For example, for my current setup this upgrade will break my application as on my private pypi server I have pycryto compiled for AIX/HPUX/etc but I don't have cffi and cryptography.

----------

The release notes and the branch looks good... expecting the note about conch.ssh

I don't plan to use this Twisted release so leaving this for review so that another person, who cares more than me about this release, can provide a better review.

Beside the note about conch.ssh, my review is basically just general feedback to the release process.

Thanks and sorry for the noise!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

Hi Adi, the release is a month old because it took 3 prereleases, each taking a week, to iron out the regressions -- so we don't want to cherry pick any further fixes, because then it pushes it out another week, and the base we've snapshotted gets even older.

Also, the conch.ssh change changes us from an extra dependency to one we already have, hence not requiring too much of a note other than that we've made the switch, since Cryptography is already required for any TLS operations.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban commented|
|-|-|

OK. No problem.

Just a feedback. For the pre-releases, why not make them based on the latest snapshot ?
I assume that people will still test the pre-release .2 in a similar way as .1 (running their full test suite) so for me it make sense to include as many fixes as possible.

If people plan to use the latest Twisted release, the will still have to test those changes on their test suite, sooner or later.

-------

I tried Twisted 15.5.0 and it was still working with pyOpenSSL 0.13... without requiring Cryptography, but maybe I am missing something.

-----

So my latest review was mostly just general feedback to the release process.
Don't worry to much about my comments.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph set owner to @hawkowl|
|-|-|

It's a shame about the OpenSSL test failures, but given the somewhat restricted nature of their impact (and the fact that they're well understood) I think it's time to SHIP IT.

Please land and release.

(And get started on 16.1.  So much great stuff in trunk.)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46994]) Merge release-16.0.0-8210: Release Twisted 16.0.0

Author: hawkowl
Reviewers: adiroiban, glyph
Fixes: [#8210](https://github.com/twisted/twisted/issues/8210)
